File: API/RunnConnectAPI/Models/Dto/Inscripcion/BusquedaInscripcionResponse.cs
using System;

namespace RunnConnectAPI.Models.Dto.Inscripcion
{
  public class BusquedaInscripcionResponse
  {
    public int IdInscripcion { get; set; }
    public DateTime FechaInscripcion { get; set; }
    public string EstadoPago { get; set; }

    //datos del evento
    public int IdEvento { get; set; }
    public string NombreEvento { get; set; } = string.Empty;
    public string NombreCategoria { get; set; } = string.Empty;

    //talle remera
    public string? TalleRemera { get; set; }

    // Datos del Runner
    public RunnerSimpleDto Runner { get; set; } = new RunnerSimpleDto();
  }

  public class RunnerSimpleDto
  {
    public string Nombre { get; set; } = string.Empty;
    public string Apellido { get; set; } = string.Empty;
    public string Dni { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;

    public string? Telefono { get; set; }
    public string? Genero { get; set; }
    public string? Localidad { get; set; }
    public string? NombreContactoEmergencia { get; set; }
    public string? TelefonoEmergencia { get; set; }

    // Propiedad calculada util para mostrar en listas
    public string NombreCompleto => $"{Nombre} {Apellido}".Trim();



  }
}

File: app/src/main/java/com/example/runnconnect/data/response/BusquedaInscripcionResponse.java
package com.example.runnconnect.data.response;

import com.google.gson.annotations.SerializedName;

public class BusquedaInscripcionResponse {
  private int idInscripcion;
  private String fechaInscripcion;
  private String estadoPago; // "pagado", "pendiente", "cancelado"
  private String estadoEvento;
  private String talleRemera;
  private int idEvento;
  private String nombreEvento;
  private String nombreCategoria;

  @SerializedName("runner")
  private RunnerSimpleInfo runner;

  // Getters
  public int getIdInscripcion() { return idInscripcion; }
  public String getEstadoPago() { return estadoPago; }
  public String getNombreEvento() { return nombreEvento; }
  public String getEstadoEvento() {return estadoEvento; }
  public String getTalleRemera() { return talleRemera; }
  public String getNombreCategoria() { return nombreCategoria; }

  public RunnerSimpleInfo getRunner() { return runner; }
  public static class RunnerSimpleInfo {
    private String nombre;
    private String apellido;
    private String dni;
    private String email;
    private String telefono;
    private String genero;
    private String localidad;
    private String nombreContactoEmergencia;
    private String telefonoEmergencia;

    public String getNombreCompleto() { return nombre + " " + (apellido != null ? apellido : ""); }
    public String getDni() { return dni; }
    public String getEmail() { return email; }
    public String getTelefono() { return telefono; }
    public String getGenero() { return genero; }
    public String getLocalidad() { return localidad; }
    public String getNombreContactoEmergencia() { return nombreContactoEmergencia; }
    public String getTelefonoEmergencia() { return telefonoEmergencia; }
  }


}
